spec/controllers/api/v1/teachers_controller_spec.rb

```diff
@@ -38,12 +38,24 @@
 
   describe "POST #create" do
     context "with valid teacher params" do
+      let(:finish_enews_subscription) { double("finish_enews_subscription") }
+
       it "creates a new teacher" do
         old_teachers_count = Portal::Teacher.count
         post :create, teacher_params
         expect(response.status).to eq(201)
         expect(Portal::Teacher.count).to eq(old_teachers_count + 1)
       end
+
+      it "creates a new teacher by SSO" do
+        user = create(:user)
+        allow(controller).to receive(:current_user) { user }
```

Comment:
```suggestion
        sign_in user
```
Instead of mocking the current user. Most other controllers sign in the user.

Reply:
@scytacki I see, thanks. I've just tried that though and the test fails. It's gone back to saying that current_user is nil.

Reply:
I think sign_in won't work if the user isn't confirmed so if you do `create(:confirmed_user)` instead that might fix it.